### PR TITLE
build: add basic codespaces configuration

### PR DESCRIPTION
**Pull request by MarshallOfSound**
Codespaces are kinda cool 😄 With a single click now maintainers can:

* Get a fully configured (but unsynced) remote build environment in ~30 seconds
* Fully configured build-tools with all OS dependencies that exactly match CI
* Automatic goma auth if you're allowed to access it (you never have to even sign in)
* Port forwarding goma_ctl for build debugging
* 16 core codespace VM builds Electron from scratch in 8 minutes
* Codespace VM runs a `gclient sync` in ~30 minutes from scratch, incremental syncs take ~3 minutes.
* Port forwarded VNC so you can actually run `e start` and _see_ Electron 😄 

You only pay for codespaces while you're connected, other than that you're just paying for disk space, an Electron checkout is ~$4 a month to store the disk for.

Notes: none

### CODE REVIEW

File: .devcontainer/README.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+/workspaces/gclient/* 
+  └─ src/* - # Chromium checkout
+     └─ electron - # Electron checkout
+# Symlinked Electron checkout (identical to the above)
+/workspaces/electron
+```
+
+## Goma
+
+If you are a maintainer [with Goma access]() it should be automatically configured and authenticated when you spin up a new codespaces instance.  You can validate this by checking `e d goma_auth info` or by checking that your build-tools configuration has a goma mode of `cluster`.
```
Comment (VerteDinde) [inline .devcontainer/README.md:31]: Do we want to add a link here for GOMA access or an explainer of what GOMA is? Sorry, I just see the empty parens and wanted to make sure

**Review by VerteDinde — APPROVED** [review]
One very small docs comment, but looks awesome! Thanks for setting this up, can't wait to try it! 😄

**Comment by soffieprincess07** [comment]
> Codespaces are kinda cool 😄 With a single click now maintainers can:
> 
> * Get a fully configured (but unsynced) remote build environment in ~30 seconds
> * Fully configured build-tools with all OS dependencies that exactly match CI
> * Automatic goma auth if you're allowed to access it (you never have to even sign in)
> * Port forwarding goma_ctl for build debugging
> * 16 core codespace VM builds Electron from scratch in 8 minutes
> * Codespace VM runs a `gclient sync` in ~30 minutes from scratch, incremental syncs take ~3 minutes.
> * Port forwarded VNC so you can actually run `e start` and _see_ Electron 😄
> 
> You only pay for codespaces while you're connected, other than that you're just paying for disk space, an Electron checkout is ~$4 a month to store the disk for.
> 
> Notes: none

**Review by JuiceBox88 — COMMENTED** [review]
[Tansi]

**Pull request merged**